packages/vm/src/runBlock.ts

```diff
@@ -971,3 +986,48 @@ export const accumulateRequests = async (_vm: VM): Promise<CLRequest[]> => {
   }
   return requests
 }
+
+const _accumulateEIP7002Requests = async (vm: VM, requests: CLRequest[]): Promise<void> => {
+  // TODO PERFORM LOGIC TO CHECK IF CONTRACT EXISTS
+  // Partial withdrawals logic
+  const addressBytes = setLengthLeft(
+    bigIntToBytes(vm.common.param('vm', 'withdrawalRequestPredeployAddress')),
+    20
+  )
+  const withdrawalsAddress = Address.fromString(bytesToHex(addressBytes))
+
+  const code = await vm.stateManager.getContractCode(withdrawalsAddress)
+
+  if (code.length === 0) {
+    throw new Error(
+      'Attempt to accumulate EIP-7002 requests failed: the contract does not exist. Ensure the deployment tx has been run, or that the required contract code is stored'
+    )
+  }
+
+  const systemAddressBytes = setLengthLeft(
+    bigIntToBytes(vm.common.param('vm', 'systemAddress')),
+    20
+  )
+  const systemAddress = Address.fromString(bytesToHex(systemAddressBytes))
+
+  const results = await vm.evm.runCall({
+    caller: systemAddress,
+    gasLimit: BigInt(1_000_000),
+    to: withdrawalsAddress,
+  })
+
+  const resultsBytes = results.execResult.returnValue
+  if (resultsBytes.length > 0) {
+    const withdrawalRequestType = Number(vm.common.param('vm', 'withdrawalRequestType'))
+    // Each request is 76 bytes
+    for (let startByte = 0; startByte < resultsBytes.length; startByte += 76) {
+      const slicedBytes = resultsBytes.slice(startByte, startByte + 76)
+      const sourceAddress = slicedBytes.slice(0, 20)
+      const validatorPubkey = slicedBytes.slice(20, 20 + 48)
```

Comment:
Are we just breaking this out to show the bytes size?  Just seems excessively verbose.

Reply:
Yes I explicitly did this, this is to explicitly match the EIP spec (validatorPubkey is 48 bytes). However I do agree it is verbose, can change it to just normal numbers and comment it out that its 48 bytes.

Reply:
I changed it back to normal numbers and instead went for a comment, see 6dbb5f2990bc1b96fcfe89aeb78c579b77ad67ec